### PR TITLE
Redesign trace search page

### LINKED ISSUE #2635: Use friendly timestamp like old UI
**Feature:**
Labels like `less than a minute ago` were good to catch when something was recent over some trace that was old (mostly interesting when you have recent traces you wanna see). Also having labels for `x minutes ago` feels more natural to read.

**Rational**
It was in old UI and I think it was good.
image (2)

**Prior Art**

image (1)

Ping @kirkov

### LINKED ISSUE #2685: Remove trace preview feature
Many may recall we had a trace preview feature in the search results screen. This I believe was an idea borrowed from jaeger. As discussed with @tacigar this is probably not a worthwhile feature.

For example, it appears to solve the trace comparison use case but it doesn't. For example, it is unlikely you will have two traces near each-other in the results and even if you did, the scrolling up and down wildly through expanded traces thing isn't that effective.

Moreover, there's recent work in the UI to share data between the trace list and the trace view page, which obviates network calls. In other words, as long as navigation such as "back" works properly, there's no reason to overload the screen, especially for performance reasons.

This should buy back code and complexity for better purposes. For example, right now no screen is ideal for comparing multiple traces. However, we could build a compare screen and let people select into it.

cc @openzipkin/ui @narayaruna

### PR DESCRIPTION
**Pull request by tacigar**
Redesign trace search page by using Material-ui components.

* Remove trace preview feature (please see #2685)
* Support the feature of filtering by service name
* Improve whole design

## OLD DESIGN:

old-design

## NEW DESIGN:

new-design-1

new-design-2

fixes #2685 
fixes #2635

### CODE REVIEW

**Comment by tacigar** [comment]
Added service filter.

ServiceFilter

**Comment by tacigar** [comment]
Added result table.

Table

**Comment by codefromthecrypt** [comment]
Is this still WIP?

**Comment by tacigar** [comment]
Yes, this is still WIP.
I noticed that I should do #2677 first...

**Comment by tacigar** [comment]
I'll add unit tests.

**Comment by tacigar** [comment]
Current state:

search-page

**Comment by codefromthecrypt** [comment]
cool client side service filter!

**Comment by tacigar** [comment]
@adriancole @zeagord 
Please review 🙇

File: zipkin-lens/src/components/DiscoverPage/TracesTab/TracesTab.jsx [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+  );
+};
+
+TracesTab.propTypes = propTypes;
+
+const mapStateToProps = state => ({
+  traceSummaries: state.traces.traceSummaries,
+});
+
+// React-Redux already provides two hooks API, useSelector and useDispatch.
```
Comment (zeagord) [inline zipkin-lens/src/components/DiscoverPage/TracesTab/TracesTab.jsx:91]: Makes sense

**Review by zeagord — APPROVED** [review]
Looks good.

**Comment by codefromthecrypt** [comment]
@tacigar are the screen shots in the description latest?

**Comment by tacigar** [comment]
@adriancole yes

**Comment by codefromthecrypt** [comment]
can you show trace detail screen? I wonder if root span name is there also

**Comment by codefromthecrypt** [comment]
epic thanks. cant wait to play

**Comment by tacigar** [comment]
> can you show trace detail screen? I wonder if root span name is there also

Sorry, not yet 😭

**Comment by codefromthecrypt** [comment]
no problem. as @zeagord advises best to do small changes when we can. This is a good foundation for small change later. excellent work

**Pull request merged**